distkeras/workers.py

```diff
@@ -110,20 +129,24 @@ def start_prefetching_thread(self, iterator):
         self.prefetching_thread.start()
 
     def prefetching(self):
-        try:
-            while self.is_prefetching:
-                if self.mini_batches.qsize() < self.max_mini_batches:
-                    batch = [next(self.iterator) for _ in range(self.batch_size)]
-                    iterator_copies = tee(batch, self.num_inputs + 1)
-                    feature_iterators = iterator_copies[:-1]
-                    label_iterator = iterator_copies[-1]
-                    X = [np.asarray([x[self.features_column[i]] for x in iterator]) 
-                        for i, iterator in enumerate(feature_iterators)]
-                    Y = np.asarray([x[self.label_column] for x in label_iterator])
-                    self.mini_batches.put([X, Y])
-        except Exception as e:
-            print(e)
-            self.is_prefetching = False
+        partition_iterators_all_epochs = tee(self.iterator, self.num_epoch)
```

Comment:
Looks solid. However, was the wait induces by the unionAll really that high? I mean, Spark is just constructing DAG nodes right?